src/boot_O2_g3/fault.c

```diff
@@ -564,354 +695,440 @@ void Fault_DrawMemDump(u32 pc, u32 sp, u32 unk0, u32 unk1) {
+        pc += sizeof(u32);
     }
 
-end:
+done:
     *spPtr = sp;
     *pcPtr = pc;
     *raPtr = ra;
 }
 
+/**
+ * Draws the stack trace page contents for the specified thread
+ */
 void Fault_DrawStackTrace(OSThread* t, u32 flags) {
-    s32 y;
-    u32 sp;
-    u32 ra;
-    u32 pc;
+    s32 line;
+    uintptr_t sp = t->context.sp;
+    uintptr_t ra = t->context.ra;
+    uintptr_t pc = t->context.pc;
     u32 pad;
-    u32 convertedPc;
-
-    sp = t->context.sp;
-    ra = t->context.ra;
-    pc = t->context.pc;
+    uintptr_t addr;
 
     Fault_FillScreenBlack();
-    FaultDrawer_DrawText(0x78, 0x10, "STACK TRACE");
-    FaultDrawer_DrawText(0x24, 0x18, "SP       PC       (VPC)");
+    FaultDrawer_DrawText(120, 16, "STACK TRACE");
+    FaultDrawer_DrawText(36, 24, "SP       PC       (VPC)");
 
-    for (y = 1; (y < 22) && (((ra != 0) || (sp != 0)) && (pc != (u32)__osCleanupThread)); y++) {
-        FaultDrawer_DrawText(0x24, y * 8 + 24, "%08x %08x", sp, pc);
+    for (line = 1; (line < 22) && (((ra != 0) || (sp != 0)) && (pc != (uintptr_t)__osCleanupThread)); line++) {
+        FaultDrawer_DrawText(0x24, line * 8 + 24, "%08x %08x", sp, pc);
 
         if (flags & 1) {
-            convertedPc = (u32)Fault_ConvertAddress((void*)pc);
-            if (convertedPc != 0) {
-                FaultDrawer_Printf(" -> %08x", convertedPc);
+            addr = Fault_ConvertAddress(pc);
+            if (addr != 0) {
+                FaultDrawer_Printf(" -> %08x", addr);
             }
         } else {
             FaultDrawer_Printf(" -> ????????");
         }
 
-        Fault_FindNextStackCall(&sp, &pc, &ra);
+        Fault_WalkStack(&sp, &pc, &ra);
     }
 }
 
-void osSyncPrintfStackTrace(OSThread* t, u32 flags) {
-    s32 y;
-    u32 sp;
-    u32 ra;
-    u32 pc;
-    u32 convertedPc;
-
-    sp = t->context.sp;
-    ra = t->context.ra;
-    pc = t->context.pc;
+void Fault_LogStackTrace(OSThread* t, u32 flags) {
+    s32 line;
+    uintptr_t sp = t->context.sp;
+    uintptr_t ra = t->context.ra;
+    uintptr_t pc = t->context.pc;
+    uintptr_t addr;
 
     osSyncPrintf("STACK TRACE");
     osSyncPrintf("SP       PC       (VPC)\n");
 
-    for (y = 1; (y < 22) && (((ra != 0) || (sp != 0)) && (pc != (u32)__osCleanupThread)); y++) {
+    for (line = 1; (line < 22) && (((ra != 0) || (sp != 0)) && (pc != (uintptr_t)__osCleanupThread)); line++) {
         osSyncPrintf("%08x %08x", sp, pc);
 
         if (flags & 1) {
-            convertedPc = (u32)Fault_ConvertAddress((void*)pc);
-            if (convertedPc != 0) {
-                osSyncPrintf(" -> %08x", convertedPc);
+            addr = Fault_ConvertAddress(pc);
+            if (addr != 0) {
+                osSyncPrintf(" -> %08x", addr);
             }
         } else {
             osSyncPrintf(" -> ????????");
         }
         osSyncPrintf("\n");
 
-        Fault_FindNextStackCall(&sp, &pc, &ra);
+        Fault_WalkStack(&sp, &pc, &ra);
     }
 }
 
 void Fault_ResumeThread(OSThread* t) {
```

Comment:
```suggestion
void Fault_ResumeThread(OSThread* thread) {
```
Might be good to double check I didn't miss any `OSThread* t -> OSThread* thread`